bitrise.yml

```diff
@@ -50,6 +50,8 @@ workflows:
         - ionic_version: "3.9.2"
         - ionic_username: $IONIC_USERNAME
         - ionic_password: $IONIC_PASSWORD
+        - cordova_ios_version: master
```

Comment:
@nhammond101 i think setting `cordova_ios_version` and `cordova_android_version` to the one used by the project would solve the ci error.

the versions:

```
{
    "ios": "4.4.0",
    "android": "6.2.3"
}
```

Could you please try it?

Reply:
@godrei, where are you pulling those values from?  Are those supposed to be stored in `package.json` or some other location?

Reply:
@godrei, if it's using the newer images which have XCode 9, the build will not work for iOS `4.4.0`, (which is why we are needing the change).  We have no insight into what error is causing the failure, but that's what causing all the problems for most users.  Do you have the ability to post the error somewhere visible to us?

Reply:
as @bayssmekanique said the requirement for the cordova-ios version is 4.5.2+.  To that end,  I've forked the test project and removed the cordova-console plugin (`<plugin name="cordova-plugin-console" spec="1.0.5" /> from config.xml`).  
Locally, the test workflow works (using xcode9), so I'm unsure what the issue is with the bitrise build, so any insight is greatly appreciated.

Reply:
@nhammond101 are you part of the Bitrise Community org?
If not, just let us know what's your Bitrise.io username and we'll add you to it.

In any case, this is the log of the failed build:

```
+------------------------------------------------------------------------------+
| (5) Step Test                                                                |
+------------------------------------------------------------------------------+
| id: ./                                                                       |
| version:                                                                     |
| collection: path                                                             |
| toolkit: go                                                                  |
| time: 2017-10-15T13:18:54-07:00                                              |
+------------------------------------------------------------------------------+
|                                                                              |
[36mINFO[0m[13:18:54]  * [32;1m[OK][0m Step dependency (go) installed, available. 
[34;1m(gradle) isn't installed, installing...[0m
[34;1m * [32;1m[OK][0m gradle installed[0m
[36mINFO[0m[13:19:14]  * [32;1m[OK][0m Step dependency (gradle) installed, available. 

[34;1mConfigs:[0m
- Platform: ios,android
- Configuration: release
- Target: emulator
- BuildConfig: 
- Options: 
- Username: ***
- Password: ***
- CordovaVersion: 
- Cordova Android Version: latest
- Cordova iOS Version: master
- IonicVersion: 3.9.2
- WorkDir: ./
- DeployDir: /Users/vagrant/deploy

[34;1mUpdating ionic version to: 3.9.2[0m
[32;1m$ npm "remove" "ionic"[0m
[32;1m$ npm "install" "-g" "ionic@3.9.2"[0m

[34;1mInstalling local ionic cli[0m
[32;1m$ npm "install" "ionic@3.9.2"[0m

[34;1mInstalling cordova and angular plugins[0m
[32;1m$ npm "install" "@ionic/cli-plugin-ionic-angular@latest" "@ionic/cli-plugin-cordova@latest"[0m

cordova version: [32;1m7.0.1[0m
ionic version: [32;1m3.9.2[0m

[34;1mIonic login[0m
[32;1m$ ionic login *** ***[0m
Log into your Ionic account
If you don't have one yet, create yours by running: ionic signup
[OK] You are logged in!

[34;1mBuilding project[0m
[32;1m$ ionic "cordova" "platform" "rm" "ios"[0m
> cordova platform remove ios --save
✔ Running command - done!
Removing platform ios from config.xml file...
Removing ios from cordova.platforms array in package.json
[32;1m$ ionic "cordova" "platform" "rm" "android"[0m
> cordova platform remove android --save
✔ Running command - done!
Removing platform android from config.xml file...
Removing android from cordova.platforms array in package.json
[32;1m$ https://github.com/apache/cordova-ios.git[0m
[32;1m$ ionic "cordova" "platform" "add" "https://github.com/apache/cordova-ios.git"[0m
> cordova platform add https://github.com/apache/cordova-ios.git --save
✔ Running command - done!
[32;1m$ android@latest[0m
[32;1m$ ionic "cordova" "platform" "add" "android@latest"[0m
> cordova platform add android@latest --save
✔ Running command - done!
[32;1m$ ionic "cordova" "build" "--release" "--emulator" "ios"[0m
[INFO] Running app-scripts build: --platform ios --target cordova
       
[13:23:55]  build dev started ... 
[13:23:55]  clean started ... 
[13:23:55]  clean finished in 1 ms 
[13:23:55]  copy started ... 
[13:23:55]  deeplinks started ... 
[13:23:55]  deeplinks finished in 15 ms 
[13:23:55]  transpile started ... 
[13:24:01]  transpile finished in 6.26 s 
[13:24:01]  preprocess started ... 
[13:24:01]  preprocess finished in less than 1 ms 
[13:24:01]  webpack started ... 
[13:24:01]  copy finished in 6.48 s 
[13:24:13]  webpack finished in 11.58 s 
[13:24:13]  sass started ... 
[13:24:15]  sass finished in 2.35 s 
[13:24:15]  postprocess started ... 
[13:24:15]  postprocess finished in 130 ms 
[13:24:15]  lint started ... 
[13:24:15]  build dev finished in 20.48 s 
> ionic cordova prepare
> cordova prepare
✔ Running command - done!

> cordova build ios --release --emulator
[13:24:20]  lint finished in 4.60 s 
✖ Running command - failed!
[ERROR] An error occurred while running cordova build ios --release --emulator (exit code 1):

        
        (truncated) ...     export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/Users/vagrant/.bitrise/tools:/usr/local/bin:/usr/local/sbin:/Users/vagrant/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/usr/local/opt/go/libexec/bin:/Users/vagrant/go/bin:/Users/vagrant/bitrise/tools/cmd-bridge/bin/osx:/Users/vagrant/Library/Developer/Xamarin/android-sdk-macosx/tools:/Users/vagrant/Library/Developer/Xamarin/android-sdk-macosx/tools/bin:/Users/vagrant/Library/Developer/Xamarin/android-sdk-macosx/platform-tools:/Library/Frameworks/Mono.framework/Versions/Current/Commands"
             /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator11.0.sdk -L/Users/vagrant/go/src/github.com/bitrise-community/steps-ionic-archive/_tmp/platforms/ios/build/emulator -F/Users/vagrant/go/src/github.com/bitrise-community/steps-ionic-archive/_tmp/platforms/ios/build/emulator -filelist /Users/vagrant/Library/Developer/Xcode/DerivedData/Ionic_Conference-fpgzyrbovuoahoaxeibhpjiwyjub/Build/Intermediates.noindex/Ionic\ Conference.build/Release-iphonesimulator/Ionic\ Conference.build/Objects-normal/x86_64/Ionic\ Conference.LinkFileList -Xlinker -rpath -Xlinker @executable_path/Frameworks -mios-simulator-version-min=9.0 -dead_strip -Xlinker -object_path_lto -Xlinker /Users/vagrant/Library/Developer/Xcode/DerivedData/Ionic_Conference-fpgzyrbovuoahoaxeibhpjiwyjub/Build/Intermediates.noindex/Ionic\ Conference.build/Release-iphonesimulator/Ionic\ Conference.build/Objects-normal/x86_64/Ionic\ Conference_lto.o -Xlinker -objc_abi_version -Xlinker 2 -fobjc-arc -fobjc-link-runtime -ObjC -Xlinker -sectcreate -Xlinker __TEXT -Xlinker __entitlements -Xlinker /Users/vagrant/Library/Developer/Xcode/DerivedData/Ionic_Conference-fpgzyrbovuoahoaxeibhpjiwyjub/Build/Intermediates.noindex/Ionic\ Conference.build/Release-iphonesimulator/Ionic\ Conference.build/Ionic\ Conference.app.xcent /Users/vagrant/go/src/github.com/bitrise-community/steps-ionic-archive/_tmp/platforms/ios/build/emulator/libCordova.a -weak_framework WebKit -lz -Xlinker -dependency_info -Xlinker /Users/vagrant/Library/Developer/Xcode/DerivedData/Ionic_Conference-fpgzyrbovuoahoaxeibhpjiwyjub/Build/Intermediates.noindex/Ionic\ Conference.build/Release-iphonesimulator/Ionic\ Conference.build/Objects-normal/x86_64/Ionic\ Conference_dependency_info.dat -o /Users/vagrant/Library/Developer/Xcode/DerivedData/Ionic_Conference-fpgzyrbovuoahoaxeibhpjiwyjub/Build/Intermediates.noindex/Ionic\ Conference.build/Release-iphonesimulator/Ionic\ Conference.build/Objects-normal/x86_64/Ionic\ Conference
        duplicate symbol _OBJC_CLASS_$_CDVLogger in:
             /Users/vagrant/Library/Developer/Xcode/DerivedData/Ionic_Conference-fpgzyrbovuoahoaxeibhpjiwyjub/Build/Intermediates.noindex/Ionic Conference.build/Release-iphonesimulator/Ionic Conference.build/Objects-normal/x86_64/CDVLogger.o
             /Users/vagrant/go/src/github.com/bitrise-community/steps-ionic-archive/_tmp/platforms/ios/build/emulator/libCordova.a(CDVLogger.o)
        duplicate symbol _OBJC_METACLASS_$_CDVLogger in:
             /Users/vagrant/Library/Developer/Xcode/DerivedData/Ionic_Conference-fpgzyrbovuoahoaxeibhpjiwyjub/Build/Intermediates.noindex/Ionic Conference.build/Release-iphonesimulator/Ionic Conference.build/Objects-normal/x86_64/CDVLogger.o
             /Users/vagrant/go/src/github.com/bitrise-community/steps-ionic-archive/_tmp/platforms/ios/build/emulator/libCordova.a(CDVLogger.o)
        ld: 2 duplicate symbols for architecture x86_64
        clang: error: linker command failed with exit code 1 (use -v to see invocation)
        
        ** BUILD FAILED **
        
        
        The following build commands failed:
        	Ld /Users/vagrant/Library/Developer/Xcode/DerivedData/Ionic_Conference-fpgzyrbovuoahoaxeibhpjiwyjub/Build/Intermediates.noindex/Ionic\ Conference.build/Release-iphonesimulator/Ionic\ Conference.build/Objects-normal/i386/Ionic\ Conference normal i386
        	Ld /Users/vagrant/Library/Developer/Xcode/DerivedData/Ionic_Conference-fpgzyrbovuoahoaxeibhpjiwyjub/Build/Intermediates.noindex/Ionic\ Conference.build/Release-iphonesimulator/Ionic\ Conference.build/Objects-normal/x86_64/Ionic\ Conference normal x86_64
        (2 failures)
        Error: Error code 65 for command: xcodebuild with args: -xcconfig,/Users/vagrant/go/src/github.com/bitrise-community/steps-ionic-archive/_tmp/platforms/ios/cordova/build-release.xcconfig,-workspace,Ionic Conference.xcworkspace,-scheme,Ionic Conference,-configuration,Release,-sdk,iphonesimulator,-destination,platform=iOS Simulator,name=iPhone SE,build,CONFIGURATION_BUILD_DIR=/Users/vagrant/go/src/github.com/bitrise-community/steps-ionic-archive/_tmp/platforms/ios/build/emulator,SHARED_PRECOMPS_DIR=/Users/vagrant/go/src/github.com/bitrise-community/steps-ionic-archive/_tmp/platforms/ios/build/sharedpch
        
[31;1mcommand failed, error: exit status 1[0m
|                                                                              |
+---+---------------------------------------------------------------+----------+
| [31;1mx[0m | [31;1mStep Test (exit code: 1)[0m                                      | 375 sec  |
+---+---------------------------------------------------------------+----------+
| Issue tracker: ...://github.com/bitrise-community/steps-ionic-archive/issues |
| Source: https://github.com/bitrise-community/steps-ionic-archive             |
+---+---------------------------------------------------------------+----------+
```

Reply:
@nhammond101 and @bayssmekanique guys could you please share your bitrise user names with me (either here or on bitrise.io on-site-chat), i would add you to the `Bitrise Community` organisation on bitrise, to let you check the step's build log.

@nhammond101 i updated the sample app used by the step. The update is not yet merged to the master branch, but you can access it on `update` branch. So to run tests with the updated sample app: in the step's bitrise.yml set `branch: "update"` for the git-clone step.

Reply:
sure, it's nhammond

Reply:
Mine is `bayssmekanique` as well

Reply:
Both of you are now invited on GitHub as well as on Bitrise.io :)

Reply:
@nhammond101, the error thrown is still the result of the `cordova-plugin-console`.  I see that you removed the plugin from the `config.xml`, but it persists in both `package.json` and `package-lock.json` (which I'm assuming is causing it to appear in the bundled app).

Reply:
Removing plugin from package files resolved the iOS issues.  Now running into Android version issues.

Reply:
I can't override the target SDK version as the `rm platform` command seems to be overriding the API version target.

One short-term solution would be to install the version 26 SDK to the test script:
```
    - script@1.1.4:
        inputs:
        - content: |-
            #!/bin/bash
            # fail if any commands fails
            set -e
            # debug log
            set -x

            echo y | android update sdk --no-ui --filter android-26
        title: Update Android
```